components/block/element.js

```diff
@@ -9,61 +10,119 @@ export default class Block extends LitElement {
   static styles = [style];
 
   static properties = {
-    blockId: { type: Number},
+    loading: {type: Boolean, state: true},
+    blockId: {type: String},
   };
 
-  renderMenu(){
-    return html`
-      <nav>
-        <button @click=${this.edit}>Edit</button>
-        <button @click=${this.moveUp}>Move Up</button>
-        <button @click=${this.moveDown}>Move Down</button>
-        <button @click=${this.delete}>Delete</button>
-      </nav>
-    `;
+  constructor() {
+    super();
+
+    this.loading = false;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this.model.on('sidebar_save:success', this.refresh.bind(this));
+  }
+
+  get layout() {
+    return window.parent.Core.g.layout;
+  }
+
+  get model() {
+    return (this.cached_model ||= this.layout.blocks.findWhere({
+      id: this.blockId,
+    }));
+  }
+
+  get parentModel() {
+    if (this.cached_parent_model !== undefined) return this.cached_parent_model;
```

Comment:
`if (this.cached_parent_model !== undefined)`
change to:
`if (this.cached_parent_model)`

No need to explicitly use `undefined`

Reply:
This is to distinguish between the cases where there is no value cached (`undefined`) and when a block has no parent (`null`).